lib/app/modules/settings/controllers/settings_controller.dart

```diff
@@ -279,9 +285,23 @@ class SettingsController extends GetxController {
     _saveSortedAlarmListPreference();
   }
 
+// change and save current language
+  void _saveCurrentLanguage() async {
+    await _secureStorageProvider.writeCurrentLanguage(
+      key: _currentLanguageKey,
+      currentLanguage: currentLanguage.value,
+    );
+  }
+
+  void _changeCurrentLanguage(String key) {
+    currentLanguage.value = key;
+    _saveCurrentLanguage();
+  }
+
   void updateLocale(String key) {
     final String languageCode = optionslocales[key]['languageCode'];
     final String countryCode = optionslocales[key]['countryCode'];
+    _changeCurrentLanguage(key);
     Get.updateLocale(Locale(languageCode, countryCode));
     local.value = Get.locale.toString();
     storage.writeLocale(languageCode, countryCode);
```

Comment:
Since we're already storing the locale here, is there any need to use _changeCurrentLanguage?

Reply:
ok, I will update according to review